src/shared/components/lollipopMutationPlot/LollipopMutationPlotWrapper.tsx

```diff
@@ -0,0 +1,508 @@
+import * as React from "react";
+import LollipopMutationPlot from "./LollipopMutationPlot";
+import {IMobXApplicationDataStore} from "../../lib/IMobXApplicationDataStore";
+import {Mutation} from "../../api/generated/CBioPortalAPI";
+import {LollipopSpec, DomainSpec} from "./LollipopMutationPlotNoTooltip";
+import {
+    default as getCanonicalMutationType, getLollipopMutationTypeFromCanonical, LollipopMutationType,
+    CanonicalMutationType
+} from "shared/lib/getCanonicalMutationType";
+import {remoteData} from "../../api/remoteData";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+import request from "superagent";
+import Response = request.Response;
+import {observer} from "mobx-react";
+import {computed, observable, action} from "mobx";
+import _ from "lodash";
+import {longestCommonStartingSubstring} from "../../lib/StringUtils";
+import {findFirstMostCommonElt} from "../../lib/findFirstMostCommonElt";
+import ReactDOM from "react-dom";
+import {Form, Button, FormGroup, InputGroup, ControlLabel, FormControl} from "react-bootstrap";
+import fileDownload from "react-file-download";
+import "./styles.scss";
+
+type LollipopMutationPlotWrapperProps = {
+    dataStore:IMobXApplicationDataStore<Mutation[]>;
+    entrezGeneId:number;
+    hugoGeneSymbol?:string;
+};
+
+const mutationTypePriority:{[canonicalMutationType:string]:number} = {
+    "missense": 1,
+    "inframe": 2,
+    "truncating": 4,
+    "nonsense": 6,
+    "nonstop": 7,
+    "nonstart": 8,
+    "frameshift": 4,
+    "frame_shift_del": 4,
+    "frame_shift_ins": 5,
+    "in_frame_ins": 3,
+    "in_frame_del": 2,
+    "splice_site": 9,
+    "fusion": 10,
+    "silent": 11,
+    "other": 11
+};
+
+function lollipopMutationTypeSort(typeA:CanonicalMutationType, typeB:CanonicalMutationType) {
+    const priorityA = mutationTypePriority[typeA];
+    const priorityB = mutationTypePriority[typeB];
+    if (priorityA < priorityB) {
+        return -1;
+    } else if (priorityA > priorityB) {
+        return 1;
+    } else {
+        return typeA.localeCompare(typeB);
+    }
+}
+
+const MISSENSE_COLOR = "#008000";
+const TRUNCATING_COLOR = "#000000";
+const INFRAME_COLOR = "#8B4513";
+const OTHER_COLOR = "#8B00C9";
+
+@observer
+export default class LollipopMutationPlotWrapper extends React.Component<LollipopMutationPlotWrapperProps, {}> {
```

Comment:
Oh, you are right. Actually, how about renaming the low level plot as just `LollipopPlot` and the wrapper as `LollipopMutationPlot`?

Ideally with `LollipopPlot` you can visualize any set of data grouped by codon/location. And maybe rename `Domains` to something more generic, like `XAxisOverlay` (or something nicer).